pkg/image/fetcher.go

```diff
@@ -35,6 +36,22 @@ type LayoutOption struct {
 	Sparse bool
 }
 
+type ImagePullChecker interface {
+	CheckImagePullInterval(imageID string, path string) (bool, error)
+	Read(path string) (*ImageJSON, error)
+	PruneOldImages(f *Fetcher) error
+	UpdateImagePullRecord(path string, imageID string, timestamp string) error
+	Write(imageJSON *ImageJSON, path string) error
+}
+
+func intervalPolicy(options FetchOptions) bool {
+	return options.PullPolicy == PullWithInterval || options.PullPolicy == PullHourly || options.PullPolicy == PullDaily || options.PullPolicy == PullWeekly
+}
+
+func NewPullPolicyManager(logger logging.Logger) *ImagePullPolicyManager {
```

Comment:
I think this method should return the interface `ImagePullChecker`, this will break the `ParsePullPolicy` but if you do the refactoring I suggested you will be able to call the internal methods